rajawali/src/main/java/c/org/rajawali3d/sceneview/SceneView.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    @Override
+    public boolean isDisabled() {
+        return false;
+    }
+
+
+    @GLThread
+    @Override
+    public void onFrameStart(final double deltaTime)
+            throws InterruptedException {
+        // TODO is a scene lock actually needed here?  This may all be independent of the Scene...
+        // TODO is a sceneView lock needed here?
+        //currentlyHeldReadLock = scene.acquireReadLock();
+        try {
+            // TODO frame tasks for changes to viewport and/or render passes
+            // Execute frame tasks
+            //performFrameTasks();
+
+            // TODO view-specific frame callbacks could be useful, e.g. enabling viewport animations
+        } finally {
+            if (currentlyHeldReadLock != null) {
+                currentlyHeldReadLock.unlock();
+            }
+        }
+    }
+
+    @GLThread
+    @Override
+    public void onRenderView() throws InterruptedException {
+        // TODO Lock should probably be on the whole Scene, not just its Graph
+        //currentlyHeldReadLock = scene.acquireReadLock();
+        try {
+            internalRender();
+        } finally {
+            if (currentlyHeldReadLock != null) {
+                currentlyHeldReadLock.unlock();
+            }
+        }
+    }
+
+    @Override
+    public void onFrameEnd(double deltaTime) throws InterruptedException {
+        // TODO Anything?
+    }
+
+    @RequiresReadLock
+    @GLThread
+    protected void internalRender() throws IllegalStateException {
+
+        // TODO Probably need to mimic the nextCameraLock pattern for viewport changes?
+        //updateProjectionMatrix(width, height);
+        //GLES20.glViewport(0, 0, width, height);
+
+        synchronized (nextCameraLock) {
+            // Check if we need to switch the camera, and if so, do it.
+            if (nextCamera != null) {
+                switchCamera(nextCamera);
+                nextCamera = null;
+            }
+        }
+
+        // Prepare the camera matrices
+        final Matrix4 viewMatrix = currentCamera.getViewMatrix();
+        final Matrix4 projectionMatrix = currentCamera.getProjectionMatrix();
+
+        if (projectionMatrix == null) {
+            throw new IllegalStateException("Cannot render while current camera has a null projection matrix.");
+        }
+
+        viewProjectionMatrix.setAll(projectionMatrix).multiply(viewMatrix);
+
+        // Determine which objects we will be rendering
+        final List<NodeMember> intersectedNodes = scene.getSceneGraph().intersection(currentCamera);
+
+        // still TODO: This will be an interaction point with the render pass manager. We don't want to check the
+        // intersection with the camera multiple times. One possible exception would be for shadow mapping. Probably
+        // a loop
+
+        // Fetch the current render type
+        int type = 0;
+
+        // Loop each node and draw
+        for (NodeMember member : intersectedNodes) {
```

Comment:
Dont worry about correcting this yourself. I have not finished the implementation here as what you are doing will affect it so i put it on hold. We will fix later.